File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/PencilTool.java
package com.savvasdalkitsis.gameframe.draw.model;

class PencilTool extends AbstractDrawingTool {

    @Override
    public void drawOn(Layer layer, int startColumn, int startRow, int column, int row, int color) {
        layer.getColorGrid().setColor(color, column, row);
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/draw/model/PencilTool.kt
package com.savvasdalkitsis.gameframe.feature.draw.model

internal class PencilTool : AbstractDrawingTool() {

    override fun drawOn(layer: Layer, startColumn: Int, startRow: Int, column: Int, row: Int, color: Int) {
        layer.colorGrid.setColor(color, column, row)
    }
}
